SROS2_Linux.md

```diff
@@ -85,13 +85,12 @@ These variables need to be defined in each terminal used for the demo. For conve
 
 ### Run the demo
 
-ROS2 allows you to [change DDS implementation at runtime](https://docs.ros.org/en/rolling/Guides/Working-with-multiple-RMW-implementations.html).
-This demo can be run with fastrtps by setting:
-```bash
-export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
-```
-And with Connext by setting:
+ROS 2 allows you to [change DDS implementation at runtime](https://docs.ros.org/en/rolling/Guides/Working-with-multiple-RMW-implementations.html).
+This demo can be run with FastDDS / CycloneDDS / ConnextDDS by setting the `RMW_IMPLEMENTATION` variable, e.g.:
+
 ```bash
+export RMW_IMPLEMENTATION=rmw_fastrtps_cpp  # or
```

Comment:
with `rmw_fastrtps_cpp`, this procedure does not work. (i tested with jazzy test environment.)

```bash
root@51cdd59e1f3e:~# export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
root@51cdd59e1f3e:~# export ROS_SECURITY_KEYSTORE=~/sros2_demo/demo_keystore
root@51cdd59e1f3e:~# export ROS_SECURITY_ENABLE=true
root@51cdd59e1f3e:~# export ROS_SECURITY_STRATEGY=Enforce
root@51cdd59e1f3e:~# ros2 run demo_nodes_cpp talker --ros-args --enclave /talker_listener/talker
[INFO] [1715030073.646233306] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/talker
[INFO] [1715030074.656773123] [talker]: Publishing: 'Hello World: 1'
...<snip>

root@51cdd59e1f3e:~# export RMW_IMPLEMENTATION=rmw_fastrtps_cpp
root@51cdd59e1f3e:~# export ROS_SECURITY_KEYSTORE=~/sros2_demo/demo_keystore
root@51cdd59e1f3e:~# export ROS_SECURITY_ENABLE=true
root@51cdd59e1f3e:~# export ROS_SECURITY_STRATEGY=Enforce
root@51cdd59e1f3e:~# ros2 run demo_nodes_py listener --ros-args --enclave /talker_listener/listener
[INFO] [1715030123.004486867] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
[INFO] [1715030125.668635649] [listener]: I heard: [Hello World: 52]
[INFO] [1715030126.659144765] [listener]: I heard: [Hello World: 53]

root@51cdd59e1f3e:~# export ROS_SECURITY_ENCLAVE_OVERRIDE=/talker_listener/listener
root@51cdd59e1f3e:~# ros2 daemon stop
root@51cdd59e1f3e:~# ros2 daemon status
root@51cdd59e1f3e:~# ros2 node list --no-daemon
[INFO] [1715030178.386225737] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
root@51cdd59e1f3e:~# ros2 topic list --no-daemon
[INFO] [1715030183.341342733] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
/parameter_events
/rosout
root@51cdd59e1f3e:~# ros2 topic echo /chatter
[INFO] [1715030189.509482358] [rcl]: Found security directory: /root/sros2_demo/demo_keystore/enclaves/talker_listener/listener
WARNING: topic [/chatter] does not appear to be published yet
Could not determine the type for the passed topic
```

Reply:
oh weird I dont have the same behavior. But it may be part of the daemon issue you point out.

On my end the topic does appear in the topic list and the echo works

(but ros2 node list returns an empty list which is weird)


```
$ ros2 daemon stop
The daemon has been stopped

$ ros2 node list
[INFO] [1715031598.709184801] [rcl]: Found security directory: /home/mikael/sros2_demo/demo_keystore/enclaves/talker_listener/listener

$ ros2 topic list
/chatter
/parameter_events
/rosout

$ ros2 topic echo /chatter
[INFO] [1715031753.543324314] [rcl]: Found security directory: /home/mikael/sros2_demo/demo_keystore/enclaves/talker_listener/listener
data: 'Hello World: 193'
---
data: 'Hello World: 194'
---
data: 'Hello World: 195'
---
```